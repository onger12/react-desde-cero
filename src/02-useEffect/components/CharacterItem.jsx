


export const CharacterItem = ({ name, nickname, img }) => {

  return (
    <div className="character">
        <img src={ img } alt="" />
        <div className="info">
            <p className="m0">{ name }</p>
            <p>{ nickname }</p>
        </div>
    </div>
  )
}
