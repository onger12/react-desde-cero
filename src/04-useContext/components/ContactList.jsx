

export const ContactList = ({ contacts = [] }) => {
    return (
        <ul className="list-group">
            {
                contacts.map( contact =>  (
                    <li>
                        <a 
                            className="list-group-item my-1" 
                            href='/contacts/1'
                        >
                                { contact.nombre } { contact.apellido} { contact.celular }
                        </a>
                    </li>
                ))
            }
            <li><a className="list-group-item my-1" href='/contacts/1'>Alisson</a></li>
        </ul>
    )
}
