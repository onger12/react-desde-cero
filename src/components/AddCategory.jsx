
import { useState } from "react";

export const AddCategory = ( { onAddCategory } ) => {

  const [inputValue, setInputValue] = useState('');

  const onSubmit = ( e ) => {
    e.preventDefault();

    if(inputValue.trim().length <= 1) return;

    onAddCategory( inputValue );
    setInputValue( '' );
  }

  const onChangeInput = ({ target }) => {
    setInputValue( target.value );
  }

  return (
    <form onSubmit={ onSubmit } aria-label="form">
      <input 
        aria-label="search-input"
        autoFocus
        value={ inputValue }
        placeholder="Buscar gifs..."
        onChange={ onChangeInput }
      />
    </form>
  )
}