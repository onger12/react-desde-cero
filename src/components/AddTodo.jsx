import { useState } from "react"



export const AddTodo = ( { onNewTodo, todos }) => {

    const [inputValue, setInputValue] = useState('');

    const onChangeInput = ({ target }) => {
        const { value } = target;
        setInputValue(value)
    }

    const onSubmit = ( event ) => {
        event.preventDefault()
        
        if( inputValue.trim() === '' ) return
        if( !!(todos.find( todo => todo.description === inputValue )) ) return
        
        const todo = {
            id: new Date().getTime(),
            description: inputValue,
            done: false,
        }

        onNewTodo(todo);

        setInputValue('');
    }

    return (
        <div className="col-5">
            <form onSubmit={ onSubmit }>
                <input 
                    type="text"
                    placeholder="¿Qué hay por hacer?"
                    className="form-control"
                    value={ inputValue }
                    onChange={ onChangeInput }
                    autoFocus
                />
                <button className="btn btn-outline-primary my-2">Agregar</button>
            </form>
        </div>
    )
}
