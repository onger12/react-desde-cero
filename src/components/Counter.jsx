import React from "react";
import PropTypes from 'prop-types'
import { useState } from "react";

const Counter = ({ value }) => {
    const [counter, setCounter] = useState( value )

    const handleAdd = () => setCounter( counter + 1 );
    const handleSubstract = () => setCounter( counter - 1 );
    const handleReset = () => setCounter( value );

    return (
        <>
            <h1>Counter App</h1>
            <h2> { counter } </h2>
            <button aria-label="btn-+1" onClick={ handleAdd }> +1 </button>
            <button aria-label="btn--1" onClick={ handleSubstract }> -1 </button>
            <button aria-label="btn-reset" onClick={ handleReset }> reset </button>
        </>
    )
}


Counter.propTypes = {
    value: PropTypes.number.isRequired
}

Counter.defaultProps = {
    value: 0
}

export default Counter