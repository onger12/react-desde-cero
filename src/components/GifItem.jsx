import PropTypes from 'prop-types';

export const GifItem = ( {title = 'gif', url, id} ) => {
    
  return (
    <div className="card">
        <img aria-label="gif-img" src={ url } alt={ title } />
        <p aria-label="p-title"> {title.trim() !== '' ? title : 'Expert Gif'} </p>
    </div>
  )
}

GifItem.propTypes = {
  title : PropTypes.string.isRequired,
  url : PropTypes.string.isRequired,
}