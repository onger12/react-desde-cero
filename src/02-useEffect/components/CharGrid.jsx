import { CharacterItem } from './CharacterItem';
import { useFetch } from '../../hooks/useFetch';

const url = 'https://www.breakingbadapi.com/api/characters';

export const CharGrid = () => {

    const {data, isLoading} = useFetch(url);

    return (
        <div className="char-grid">
            {
                isLoading ? 
                (<div className="alert alert-info text-center">Loading...</div>) :
                (data.map(char => <CharacterItem { ...char } />))
            }
        </div>
    )
}
