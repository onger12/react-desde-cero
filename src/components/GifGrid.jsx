import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const GifGrid = ( { category = 'Goku'} ) => {

    const { gifs, isLoading } = useFetchGifs( category );

    return (
        <>
            { isLoading ? 
                <h1>Loading { category } gifs...</h1> : 
                <>
                    <h1> { category } </h1>
                    <div className="card-grid">
                    {
                        gifs.map(gif => (
                            <GifItem
                            key={ gif.id }
                            { ...gif } />
                            ))
                        }

                    </div>
                </>
            }
        </>
    )
}


GifGrid.propTypes = {
    category : PropTypes.string.isRequired,
}