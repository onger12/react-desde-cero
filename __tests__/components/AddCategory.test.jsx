import { render, screen, fireEvent, debug } from '@testing-library/react';
import { AddCategory } from '../../src/components/AddCategory';

describe( 'Testeando <AddCategory />', () => {

    const setup = () => {
        const utils = render( <AddCategory /> );
        const input = screen.getByLabelText( 'search-input' );
        const form = screen.getByLabelText( 'form' );

        return {
            input,
            form,
            ...utils
        }
    }
    
    test('debe hacer match con el snapshot', () => {
        const { container } = setup();
        expect( container ).toMatchSnapshot();
    })
    
    test('input.value debe actualizarse con el evento onChange', () => {
        const { input } = setup();
        const value = 'Braking Bad';
        fireEvent.change( input, { target : { value } } );
        
        expect( input.value ).toBe( value );
    })

    test('el form debe enviar el input.value con el evento onSubmit', () => {
        const { input, form } = setup();
        const value = 'Braking Bad';

        fireEvent.change( input, { target : { value } } );


        console.log( input.value );
        expect( input.value ).toBe( value );
    })
})