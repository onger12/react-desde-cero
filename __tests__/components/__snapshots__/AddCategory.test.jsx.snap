// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeando <AddCategory /> debe hacer match con el snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <input
      aria-label="search-input"
      placeholder="Buscar gifs..."
      value=""
    />
  </form>
</div>
`;
